packages/block-editor/src/hooks/color.js

```diff
@@ -371,61 +455,140 @@ export function ColorEdit( props ) {
-								onGradientChange: hasGradientColor
-									? onChangeGradient
-									: undefined,
-							},
-					  ]
-					: [] ),
-				...( hasLinkColor
-					? [
-							{
-								label: __( 'Link Color' ),
-								onColorChange: onChangeLinkColor,
-								colorValue: getLinkColorFromAttributeValue(
-									allSolids,
-									style?.elements?.link?.color?.text
-								),
-								clearable: !! style?.elements?.link?.color
-									?.text,
-							},
-					  ]
-					: [] ),
-			] }
-		/>
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Colors' ) }
+				className="block-editor__hooks-colors-panel"
+			>
+				<NavigatorProvider initialPath="/">
+					<NavigatorScreen path="/">
+						<ItemGroup isBordered isSeparated>
+							{ ( hasBackgroundColor || hasGradientColor ) && (
+								<NavigationButton path="/background">
+									<HStack justify="flex-start">
+										<FlexItem>
+											<ColorIndicator
+												colorValue={
+													gradientValue ??
+													backgroundValue
+												}
+											/>
+										</FlexItem>
+										<FlexItem>
+											{ __( 'Background' ) }
+										</FlexItem>
+									</HStack>
+								</NavigationButton>
+							) }
+							{ hasTextColor && (
+								<NavigationButton path="/text">
+									<HStack justify="flex-start">
+										<FlexItem>
+											<ColorIndicator
+												colorValue={ textColorValue }
+											/>
+										</FlexItem>
+										<FlexItem>{ __( 'Text' ) }</FlexItem>
+									</HStack>
+								</NavigationButton>
+							) }
+							{ hasLinkColor && (
+								<NavigationButton path="/link">
+									<HStack justify="flex-start">
+										<FlexItem>
+											<ColorIndicator
+												colorValue={ linkColorValue }
+											/>
+										</FlexItem>
+										<FlexItem>{ __( 'Links' ) }</FlexItem>
+									</HStack>
+								</NavigationButton>
+							) }
+						</ItemGroup>
+					</NavigatorScreen>
+
+					{ ( hasBackgroundColor || hasGradientColor ) && (
+						<NavigatorScreen path="/background">
+							<ScreenHeader
+								back="/"
+								title={ __( 'Background' ) }
+							/>
+							<ColorPanel
+								enableContrastChecking={
+									// Turn on contrast checker for web only since it's not supported on mobile yet.
+									Platform.OS === 'web' &&
+									! gradient &&
+									! style?.color?.gradient
+								}
+								clientId={ props.clientId }
+								settings={ [
+									{
+										label: __( 'Background color' ),
+										onColorChange: hasBackgroundColor
+											? onChangeColor( 'background' )
+											: undefined,
+										colorValue: backgroundValue,
+										gradientValue,
+										onGradientChange: hasGradientColor
+											? onChangeGradient
+											: undefined,
+									},
+								] }
+							/>
+						</NavigatorScreen>
+					) }
+
+					{ hasTextColor && (
+						<NavigatorScreen path="/text">
+							<ScreenHeader back="/" title={ __( 'Text' ) } />
```

Comment:
These "back" links need the ability to focus the element they represent when you go back, otherwise the top one is always selected. This also happens in global styles.

Reply:
mmm, this might not be easy without introducing some kind of `#hash` support in the "path".

Reply:
cc @ciampo @diegohaz this seems pretty important for keyboard accessibility

Reply:
What if the `NavigatorScreen` had the ability to move the focus on itself (or on a children) when mounted?

Reply:
For me the only to know the element that was focused before moving to a new "screen" is:

 - Keep a reference to the old element
 - Or have an identifier for the element.

We don't render all the screens at the same time, meaning, even if keep a reference, when going back to the previous screen, new elements will be rendered so that reference is useless.

Meaning, we only have one option: add identifiers to elements and pass the identifier somehow to `navigator.push` calls.

Reply:
I see what you mean. We basically need to keep track of the last "navigator link" that was clicked.

> pass the identifier somehow

Should that be a `Ref` ?

Because the only way to trigger a navigation is via the `push` function, we will have to manually pass that identifier every time.

We should also account for that "identifier" to be `null`|`undefined` — for example, if we navigate programmatically (and not after the user clicks on a link).

Reply:
Mmm, I was hoping `ScreenHeader` would know the context it's coming from and pass it on.

Reply:
> Mmm, I was hoping ScreenHeader would know the context it's coming from and pass it on.

The problem is how to model the "context" here. How to store it and match it with the previous location when coming back.

Reply:
We have a component with a `back="/"` nested within a component with `path="/text"`, and in the main view we have a `NavigationButton` with a `path="/text"`. It seems to me we should have all the pieces to model the connection of a back button coming from a non-root path to the button that directs to that path.

Reply:
> We have a component with a back="/" nested within a component with path="/text", and in the main view we have a NavigationButton with a path="/text". It seems to me we should have all the pieces to model the connection of a back button coming from a non-root path to the button that directs to that path.

In my opinion, this solution:
- would be specific only to the sidebar, since `Navigator` doesn't rely on the `path` prop at all (it only exposes a `push` function via `useNavigator`)
- can be brittle, because we're not guaranteed that there is only one component with `path="text"` or that the `path` isn't dynamic etc

The approach proposed by Riad would potentially solve these 2 aspects:
- the feature would be added to the `push` function and to the `NavigatorContext`, and thus would be available to all consumers of the component
- using a proper "identifier" eliminates the risk of not being able to focus the correct component when navigating back (although we haven't defined exactly what type of identifier — `id` attribute, React `ref`, ...?)

I know that this is just one relatively small feature that we're discussing, but it reminds me of one of the points that I made initially when planning for `Navigator` — if we find ourselves in need of adding little by little more features, we could potentially end up with a complex and hard to maintain component. If this is the trajectory that we're onto, we should consider using a third party routing library instead.

Reply:
The concept of returning focus somewhere specifically is likely broader as well than routing

Reply:
Routing libraries usually provide an API to pass state along with route changes. It seems like that's what we need here. I agree that passing an `id` is better. If the host component is unmounted, we'll lose the `ref` object reference, so I'm not sure that would work.

Alternatively, instead of passing state to new screens, we can explore keeping hidden screens mounted (just the wrapper components) so they can get the activeElement before transitioning to a new screen and restore the focus when the screen is active again. This might be a bit complicated though.

Reply:
> we can explore keeping hidden screens mounted 

That would be a huge performance degradation for Global Styles, there's hundreds if not thousands of screens (blocks...)

Reply:
Would moving the focus on the new "screen" (instead that on the precise link) be an acceptable compromise? That would simplify the solution considerably (we'd just need to add the option for a screen to auto-focus when mounted)

Reply:
Let's move this conversation into an issue specifically for global styles, since we might not end up doing anything here on the block inspector.

Reply:
Good point. I've opened #37063

Reply:
Thanks!